platform/features/table/src/controllers/MCTTableController.js

```diff
@@ -295,27 +448,25 @@ define(
          * will be sorted before display.
          */
         MCTTableController.prototype.updateRows = function (newRows) {
-            var displayRows = newRows;
+            //Reset visible rows because new row data available.
             this.$scope.visibleRows = [];
+
             this.$scope.overrideRowPositioning = false;
 
+            //Nothing to show because no columns visible
             if (!this.$scope.displayHeaders) {
                 return;
             }
 
-            if (this.$scope.enableFilter) {
-                displayRows = this.filterRows(displayRows);
-            }
-
-            if (this.$scope.enableSort) {
-                displayRows = this.sortRows(displayRows);
-            }
-            this.$scope.displayRows = displayRows;
+            //Apply filters and sort a copy of the the new rows
+            this.filterAndSort((newRows || []).slice(0));
+            //Resize columns appropriately
```

Comment:
The shallowClone is actually duplicated in a few places-- filterRows uses filter which returns a new array, while sortRows is shallow cloning the array prior to doing an in place sort.  Should minimize number of times that a new array has to be instantiated.